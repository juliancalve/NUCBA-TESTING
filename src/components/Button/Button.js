import React from 'react'

const Button = ({ handleClick, title }) => {
    return (
        <div>
            <button onClick={handleClick}>{title}</button>
        </div>
    )
}

export default Button
