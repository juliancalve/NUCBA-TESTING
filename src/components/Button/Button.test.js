import { shallow } from "enzyme";
import Button from "./Button";

describe('Button component test', () => {

    test('Tiene que estar en el snapshot', () => {
        const wrapper = shallow(<Button />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('Tiene que contener el titulo que se le pasa por parametro', () => {
        const title = 'soy un titulo';
        const wrapper = shallow(<Button title={title} />);

        const button = wrapper.find('button');

        expect( button.text() ).toBe(title);

    });

    test('la suma tiene que coincidir', () => {

        const num1 = 3;
        const num2 = 2;
        const result = 5;

        expect( num1 + num2 ).toBe(result);
    });

    test('Checkear que la funcion se este llamando al clickear', () => {
        const title = 'titulo';
        const miFuncion = jest.fn();
        const wrapper = shallow(<Button title={title} handleClick={miFuncion} />);

        const button = wrapper.find('button');
        button.simulate('click');

        expect(miFuncion).toBeCalledTimes(1);
    });

});