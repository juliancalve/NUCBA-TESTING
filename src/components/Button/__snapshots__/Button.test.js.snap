// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test Tiene que estar en el snapshot 1`] = `
<div>
  <button />
</div>
`;
